From c74761524895b1db5f048831579d036ad5f4768a Mon Sep 17 00:00:00 2001
From: torque <torque@1>
Date: Sun, 13 Sep 2015 15:57:25 -0700
Subject: [PATCH] fontselect: replace iconv with WideCharToMultiByte.

---
 libass/ass_fontselect.c | 23 +++++------------------
 1 file changed, 5 insertions(+), 18 deletions(-)

diff --git a/libass/ass_fontselect.c b/libass/ass_fontselect.c
index 245ec38..9d4f7bc 100644
--- a/libass/ass_fontselect.c
+++ b/libass/ass_fontselect.c
@@ -35,7 +35,7 @@
 #include FT_FREETYPE_H
 #include FT_SFNT_NAMES_H
 #include FT_TRUETYPE_IDS_H
-#include <iconv.h>
+#include <Windows.h>
 
 #include "ass_utils.h"
 #include "ass.h"
@@ -721,18 +721,11 @@ get_font_info(FT_Library lib, FT_Face face, ASS_FontProviderMetaData *info)
     int slant, weight;
     char *fullnames[MAX_FULLNAME];
     char *families[MAX_FULLNAME];
-    iconv_t utf16to8;
 
     // we're only interested in outlines
     if (!(face->face_flags & FT_FACE_FLAG_SCALABLE))
         return 0;
 
-    // scan font names
-    utf16to8 = iconv_open("UTF-8", "UTF-16BE");
-
-    if (utf16to8 == (iconv_t)-1)
-        goto error;
-
     for (i = 0; i < num_names; i++) {
         FT_SfntName name;
 
@@ -743,15 +736,14 @@ get_font_info(FT_Library lib, FT_Face face, ASS_FontProviderMetaData *info)
                 (name.name_id == TT_NAME_ID_FULL_NAME ||
                  name.name_id == TT_NAME_ID_FONT_FAMILY)) {
             char buf[1024];
-            char *bufptr = buf;
-            size_t inbytes = name.string_len;
+            size_t inchars  = name.string_len / 2;
             size_t outbytes = 1024;
 
-            if (iconv(utf16to8, (char**)&name.string, &inbytes, &bufptr,
-                        &outbytes) == (size_t)-1)
+            int written = WideCharToMultiByte(CP_UTF8, 0, name.string, inchars, buf, outbytes - 1, NULL, NULL);
+            if ( written == 0 )
                 continue;
+            buf[written] = '\0';
 
-            *bufptr = '\0';
 
             if (name.name_id == TT_NAME_ID_FULL_NAME) {
                 fullnames[num_fullname] = strdup_trimmed(buf);
@@ -769,8 +761,6 @@ get_font_info(FT_Library lib, FT_Face face, ASS_FontProviderMetaData *info)
         }
 
     }
-    iconv_close(utf16to8);
-    utf16to8 = (iconv_t)-1;
 
     // check if we got a valid family - if not use whatever FreeType gives us
     if (num_family == 0 && face->family_name) {
@@ -811,9 +801,6 @@ get_font_info(FT_Library lib, FT_Face face, ASS_FontProviderMetaData *info)
     return 0;
 
 error:
-    if (utf16to8 != (iconv_t)-1)
-        iconv_close(utf16to8);
-
     for (i = 0; i < num_family; i++)
         free(families[i]);
 
-- 
2.2.1

